lib/findCurrentImports.js

```diff
@@ -1,69 +1,131 @@
-    skipped += skipSection;
+function convertToImportStatement(node: Object): ImportStatement {
+  if (node.type === 'ImportDeclaration') {
+    const defaultSpecifier = node.specifiers.find((spec: Object): Object =>
+        spec.type === 'ImportDefaultSpecifier');
+    return new ImportStatement({
+      declarationKeyword: 'import',
+      defaultImport: defaultSpecifier ? defaultSpecifier.local.name : undefined,
+      hasSideEffects: node.specifiers.length === 0,
+      namedImports: node.specifiers.map((spec: Object): Array<?String> => {
+        if (spec.type !== 'ImportSpecifier') {
+          return undefined;
+        }
+        return spec.local.name;
+      }).filter(Boolean),
+      path: node.source.value,
+    });
   }
 
-  // We don't want to skip over blocks that are only whitespace
-  if (xRegExp.test(skipped, xRegExp('^\\s+$', 's'))) {
-    scanner.reset();
-  } else {
-    const count = (skipped.match(/\n/g) || []).length;
-    importsStartAt += count;
+  if (node.type === 'ExpressionStatement') {
+    if (!node.expression.callee) {
+      return undefined;
+    }
+    if (node.expression.callee.name !== 'require') {
+      return undefined;
+    }
+
+    if (node.expression.arguments.length !== 1) {
+      return undefined;
+    }
+
+    return new ImportStatement({
+      hasSideEffects: true,
+      importFunction: 'require',
+      path: node.expression.arguments[0].value,
+    });
   }
 
-  const imports = new ImportStatements(config);
-  let potentialImport;
-  while (potentialImport = scanner.scan(xRegExp('(^\\s*\\n)*^.*?;\\n', 's'))) {
-    const importStatement = ImportStatement.parse(potentialImport.trim());
-    if (!importStatement) {
-      break;
+  if (node.type === 'VariableDeclaration') {
+    if (!node.declarations || node.declarations.length > 1) {
+      return undefined;
+    }
+
+    const declaration = node.declarations[0];
+    if (declaration.init.type !== 'CallExpression') {
+      return undefined;
     }
 
-    imports.push(importStatement);
-    const count = (potentialImport.match(/\n/g) || []).length;
-    newlineCount += count;
+    const defaultImport = declaration.id.type === 'Identifier' ?
+      declaration.id.name : undefined;
+
+    const namedImports = declaration.id.type === 'ObjectPattern' ?
+      declaration.id.properties.map((p: Object): String => p.value.name) : undefined;
+
+    return new ImportStatement({
+      declarationKeyword: node.kind,
+      defaultImport,
+      hasSideEffects: false,
+      importFunction: declaration.init.callee.name,
+      namedImports,
+      path: declaration.init.arguments[0].value,
+    });
   }
+  return undefined;
+}
 
-  return {
-    imports,
+function parse(currentFileContent: String): Object {
+  // eslint-disable-next-line global-require
+  const babylon = require('babylon');
+  try {
+    return babylon.parse(currentFileContent, {
+      allowImportExportEverywhere: true,
+      plugins: [
+        'jsx',
+        'flow',
```

Comment:
It doesn't happen under the hood. I worry that we'll end up in the same version mismatch issues that we have today with eslint. I'd like to avoid looking for local config and plugins until we have a report of this being a problem.